### PR TITLE
Fix ReloadAmmoPool.ResetFire ignoring multipliers

### LINKED ISSUE #17399: ReloadAmmoPool.ResetOnFire ignores ReloadAmmoDelayMultipliers.

### PR DESCRIPTION
**Pull request by reaperrr**
Fixes #17399.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]

**Comment by reaperrr** [comment]
Updated, applied fix to creation as well. It's possible there already are active modifiers at that point, after all.

**Comment by abcdefg30** [comment]
My :+1: still stands.

File: OpenRA.Mods.Common/Traits/ReloadAmmoPool.cs [hunk 1]
```diff
@@ -58,14 +58,14 @@ protected override void Created(Actor self)
 		{
 			ammoPool = self.TraitsImplementing<AmmoPool>().Single(ap => ap.Info.Name == Info.AmmoPool);
 			modifiers = self.TraitsImplementing<IReloadAmmoModifier>().ToArray();
-			remainingTicks = Info.Delay;
+			remainingTicks = Util.ApplyPercentageModifiers(Info.Delay, modifiers.Select(m => m.GetReloadAmmoModifier()));
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ReloadAmmoPool.cs]: The status of conditional traits aren't guaranteed to be correct until after `Created` has been completed - this needs to be deferred to either a FrameEndTask or the first tick.

**Comment by reaperrr** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/078c10034afc33cc7164754959bc263f684d77e4

**Pull request merged**